src/server/cmd/pachd/main.go

```diff
@@ -300,23 +301,35 @@ func doFullMode(appEnvObj interface{}) error {
 	}
 
 	healthServer := health.NewHealthServer()
-	return grpcutil.Serve(
-		func(s *grpc.Server) {
-			pfsclient.RegisterAPIServer(s, pfsAPIServer)
-			pfsclient.RegisterObjectAPIServer(s, blockAPIServer)
-			ppsclient.RegisterAPIServer(s, ppsAPIServer)
-			cache_pb.RegisterGroupCacheServer(s, cacheServer)
-			authclient.RegisterAPIServer(s, authAPIServer)
-			healthclient.RegisterHealthServer(s, healthServer)
-		},
-		grpcutil.ServeOptions{
-			Version:    version.Version,
-			MaxMsgSize: grpcutil.MaxMsgSize,
-		},
-		grpcutil.ServeEnv{
-			GRPCPort: appEnv.Port,
-		},
-	)
+
+	httpServer, err := pfs_server.NewHTTPServer(address, []string{etcdAddress}, appEnv.PFSEtcdPrefix, blockCacheBytes)
```

Comment:
I double checked, and no ... the pprof API won't be served from the service we're adding. We do still need the separate `ListenAndServe` call